### DISCUSSION TITLE
Allow restricting the Access-Control-Allow-Origin header to a specific origin

### DISCUSSION THREAD

**Discussion by chipilov** [supabase/supabase#7038] [Feature Requests] [answered]
It seems that Supabase currently handles CORS by using a wildcard for the Access-Control-Allow-Origin (i.e. any origin can send requests to a Supabase project).

I think it would be beneficial to allow restricting the access only to a specific origin via the settings in the Dashboard (similar to how redirects can be restricted to specific URLs in the Auth settings of the dashboard).

**Comment by logemann** [comment]
If you want to restrict access on a network level, it looks like supabase team is working on it: 
https://supabase.com/docs/reference/cli/supabase-network-restrictions

Thats of course not CORS but with more or less the same goal.

**Comment by farsightsoftware** [comment]
@chipilov This is for the out-of-the-box database API? (I ask because you can control the CORS options for edge functions, but I'm not even sure they had edge functions when you asked this, so... :-) )

It would indeed be very handy to be able to directly control not just the origins for the CORS responses for the DB API, but other things (like `Access-Control-Max-Age`, to reduce `OPTIONS` calls).

**Reply by chipilov** [reply]
Yep, I am talking about the out-of-the-box database API via PostgREST.

**Comment by steve-chavez** [comment, marked as answer]
For the Data API (PostgREST) this can already be done with:

```sql
ALTER ROLE authenticator SET pgrst.server_cors_allowed_origins = 'https://foo.example';
NOTIFY pgrst, 'reload config';
```

https://postgrest.org/en/v12/references/configuration.html#server-cors-allowed-origins

Note that this doesn't control other services, like Auth.

**Reply by Sockitos** [reply]
Can anyone confirm this works? 
I've used this example and ran some tests, and it still always returns "Access-Control-Allow-Origin: *".
Could Cloudflare be overriding this header?

**Reply by steve-chavez** [reply]
@Sockitos You're right, while postgREST has the feature, currently the proxy in Supabase is overriding the header. Could you file a support ticket?

**Reply by Sockitos** [reply]
Ticket created: https://github.com/supabase/supabase/issues/42033

**Comment by fahimalizain** [comment]
Is there any way to control `Access-Control-Expose-Headers`?
